src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -579,19 +581,18 @@ export class DynamicFacet extends Component implements IDynamicFacet {
     const response = index !== -1 ? results.facets[index] : null;
     this.position = index + 1;
 
-    this.onQueryResponse(response);
+    response ? this.onQueryResponse(response) : this.onNoValues();
     this.updateQueryStateModel();
     this.header.hideLoading();
-    this.values.render();
     this.updateAppearance();
   }
 
-  private onQueryResponse(response?: IFacetResponse) {
-    if (response) {
-      this.moreValuesAvailable = response.moreValuesAvailable;
-      return this.values.createFromResponse(response);
-    }
+  private onQueryResponse(response: IFacetResponse) {
+    this.moreValuesAvailable = response.moreValuesAvailable;
+    this.values.createFromResponse(response);
+  }
```

Comment:
Maybe something like `onValuesFound` could be more specific now that you've extracted the condition to the caller function.

Reply:
still it's receiving the query response as parameter and then builds the values from it

Reply:
As both functions are being called in the same ternary, it seems to me the naming of them must be at the same level of abstraction. `onQueryResponse` sounds more like plumbing related, and `onNoValues` sounds more business or domain related.

But no big deal ;-)